### ISSUE TITLE
Point at infinity not emplemented

### ISSUE THREAD

**Issue by shikuk** [AntonKueltz/fastecdsa#21]
Operands with resulting point at infinity fails to 
`ValueError: (x, y) coordinates are not on curve <test_curve>`  or simply does wrong (0*G or N*G )
```
print ("G =  %s" % (G))
print ("0*G = %s" % (0*G))
print ("G - G = %s" % (G - G))
```

```
G =  X: 0x500
Y: 0x823
(On curve <test_curve>)
0*G = X: 0x500
Y: 0x823
(On curve <test_curve>)
    print ("G - G = %s" % (G - G))
  File "build\bdist.win32\egg\fastecdsa\point.py", line 99, in __sub__
  File "build\bdist.win32\egg\fastecdsa\point.py", line 74, in __add__
  File "build\bdist.win32\egg\fastecdsa\point.py", line 31, in __init__
ValueError: (x, y) coordinates are not on curve <test_curve>
```
May be need to define Infinity point and add check for this to "is_point_on_curve"

**Comment by shikuk** [comment]
Tested, work as expected. Thank You!!!

**Comment by shikuk** [comment]
Addition of points R + (-R)  fails
m - random. Checked at random curves. Issue catched by implementation of Miller's algorithm
```
# n = #E(P192)
n = 6277101735386680763835789423176059013767194773182842284081
m = 0xAA5E28D6A97A2479A65527F7290311A3624D4CC0FA157
nm = n - m
R = m * P192.G
negR = nm * P192.G
FAIL = R + negR 
```

-> 
`ValueError: coordinates are not on curve <P192>`
Expected 'POINT AT INFINITY'  to be here.

**Comment by AntonKueltz** [comment]
Good catch, not sure why this didn't already have test coverage. Fixed in commit 3a7eb8aefd19d59c9ce35c09be7f1897ed623918.

```python
In [1]: from fastecdsa.curve import P192

In [2]: n = 6277101735386680763835789423176059013767194773182842284081
   ...: m = 0xAA5E28D6A97A2479A65527F7290311A3624D4CC0FA157
   ...: nm = n - m
   ...: R = m * P192.G
   ...: negR = nm * P192.G
   ...: R + negR
Out[2]: <POINT AT INFINITY>
```
